tests/integration/test_data_collector_reliability.py

```
@@ -0,0 +1,422 @@
+            result = self.client.get_grouped_daily("2024-01-01")
+            assert len(result) == 1
+            assert call_count == 2
+
+    @pytest.mark.asyncio
+    async def test_async_retry_mechanisms(self):
+        """Test async retry mechanisms in fundamentals client."""
+        fundamentals_client = PolygonFundamentalsClient()
+
+        # Mock async network failure followed by success
+        call_count = 0
+        async def mock_async_request(*args, **kwargs):
```

Ruff (0.14.3)

Unused function argument: `args`

(ARG001)